api/targets/credential_utils.go

```diff
@@ -0,0 +1,19 @@
+package targets
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"strings"
+)
+
+// DecodeJsonSecret converts base64 encoded json strings into a go map.
+// Numerical values in the map are of type json.Number.
+func DecodeJsonSecret(in string) (map[string]interface{}, error) {
```

Comment:
This is only used in the `internal` package right now. Can you move it somewhere in there, or copy the code there, so that isn't an exported/public part of the API since the API doesn't need it yet (and maybe ever)?

Reply:
moved it in with the one test that uses it.